source/Calamari.Shared/Integration/Packages/Download/S3PackageDownloader.cs

```diff
@@ -98,12 +100,16 @@ string BuildFileName(string prefix, string version, string extension)
                                          .GetAwaiter()
                                          .GetResult();
 #endif
+                            if (fileExists)
+                            {
+                                knownExtension = knownFileExtensions[i];
+                            }
                         }
 
                         if (!fileExists)
                             throw new Exception($"Unable to download package {packageId} {version}: file not found");
 
-                        var localDownloadName = Path.Combine(cacheDirectory, PackageName.ToCachedFileName(packageId, version, "." + Path.GetExtension(fileName)));
+                        var localDownloadName = Path.Combine(cacheDirectory, PackageName.ToCachedFileName(packageId, version, knownExtension));
```

Comment:
To search for a package we iterate over each known extension type. Here we're changing to use the extension type that was used to download the package. 

Previously this included an additional . in the path, meaning we'd set the downloaded package to `*..zip`. This also avoids issues confusing these file extensions ".tar.gz", ".tar.bz2", ".tar.gz", ".tgz", ".tar.bz"
![image](https://github.com/OctopusDeploy/Calamari/assets/101079287/0aa75347-b0ec-47bc-9798-85854cb05d4c)